### DISCUSSION TITLE
Upgrade the GUI

### DISCUSSION THREAD

**Discussion by jsuto** [jsuto/piler#30] [Ideas]
The piler GUI has become outdated by now. It uses an old version of bootstrap and jquery. The security analysis performed by github CodeQL hurts my pride, so I've decided it's high time to upgrade the GUI.

I use the term "upgrade" in a quite loose fashion in this context, because it can mean simply to use the latest stable bootstrap and jquery (really?) or even to rewrite the GUI from scratch using a very different tech stack.

### Keep PHP as the backend language
- upgrade bootstrap and jquery
- upgrade bootstrap, but replace jquery with something else, eg. vue 3
- replace bootstrap+jquery with something else
- ...

### Replace PHP
- some java based stuff, eg. kotlin
- a Python based framework
- node.js
- ...

Let me know what you think.

**Comment by adn77** [comment]
I have been using Piler for a long time because its different pieces are
* elegant
* lightweight
* using the best suited technology for each task
* quite easy to understand

Considering how often a user actually searches the mail archive, the look and feel of the frontend should not be the most important thing to look at.

There is a lot of archive funtionality in the PHP-backend code, especially the rights management (simple, yet effective). If you were to replace those, it would certainly open a number of rounds of testing.

Finally, having seen the increasing number of supply chain vulnerabilities of the past couple of months/years, I would really hate to see the **lightweight** nature of the GUI gone.

Without having seen the CodeQL, I'd opt for bootstrap+jquery 🙂

**Comment by MalakymR** [comment]
I'm with @adn77 on this. The interface doesn't need much more, its clean, I would say it still looks "modern"... whatever that means these days.

Functionality should trump looks for an archival tool.

I would upgrade bootstrap+jquery then consider replacing jquery.

**Comment by jsuto** [comment]
OK. In the meantime I've been working on a PR that upgrades bootstrap and jquery.

There will be some optimizations as well:
- dropped jquery-ui, I think we can live without it
- also dropping any css + javascript from the repo that can be referenced from a CDN
- replace roundcube's splitter with something simpler

**Comment by jsuto** [comment]
PR #36 implements the updated bootstrap and jquery
